### PR TITLE
Fixes #32219 - Check if SPDC settings are available

### PR DESCRIPTION
**Pull request by adamruzicka**
before trying to use them

### CODE REVIEW

File: lib/foreman_remote_execution_core.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def self.validate_ssh_log_level!
 
     current = if defined?(::Proxy::SETTINGS)
                 ::Proxy::SETTINGS.log_level.to_s.downcase
-              elsif defined?(SmartProxyDynflowCore)
+              elsif defined?(SmartProxyDynflowCore::SETTINGS)
```
Comment (ares) [inline lib/foreman_remote_execution_core.rb:37]: would it make sense to do rather ` defined?(SmartProxyDynflowCore) && defined?(SmartProxyDynflowCore::SETTINGS)`

Comment (adamruzicka) [reply lib/foreman_remote_execution_core.rb:37]: Is it necessary? What would be the scenario where your suggestion would behave differently?

```
2.7.0 :001 > defined?(::SmartProxyDynflowCore)
 => nil 
2.7.0 :002 > defined?(::SmartProxyDynflowCore::SETTINGS)
 => nil
```

Comment (ekohl) [reply lib/foreman_remote_execution_core.rb:37]: @ares ping?

Comment (ares) [reply lib/foreman_remote_execution_core.rb:37]: ah true, ignore me

**Review by ares — APPROVED** [review]
Thanks @adamruzicka, merging!

**Pull request merged**